packages/lodestar/src/eth1/stream.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * @module eth1
+ */
+
+import {AbortSignal} from "abort-controller";
+import {IDepositEvent, IBatchDepositEvents, IEth1Block, IEth1Provider, IEth1StreamParams} from "./interface";
+import {groupDepositEventsByBlock} from "./utils/groupDepositEventsByBlock";
+import {optimizeNextBlockDiffForGenesis} from "./utils/optimizeNextBlockDiffForGenesis";
+import {sleep} from "../util/sleep";
+import {ErrorAborted} from "../util/errors";
+
+/**
+ * Phase 1 of genesis building.
+ * Not enough validators, only stream deposits
+ * @param signal Abort stream returning after a while loop cycle. Aborts internal sleep
+ */
+export async function* getDepositsStream(
+  fromBlock: number,
+  provider: IEth1Provider,
+  params: IEth1StreamParams,
+  signal?: AbortSignal
+): AsyncGenerator<IBatchDepositEvents> {
+  fromBlock = Math.max(fromBlock, provider.deployBlock);
+
+  while (true) {
+    const remoteFollowBlock = await getRemoteFollowBlock(provider, params);
+    const toBlock = Math.min(remoteFollowBlock, fromBlock + params.MAX_BLOCKS_PER_POLL);
+    const logs = await provider.getDepositEvents(fromBlock, toBlock);
+    for (const batchedDeposits of groupDepositEventsByBlock(logs)) {
+      yield batchedDeposits;
+    }
+
+    fromBlock = toBlock;
+
+    if (signal && signal.aborted) throw new ErrorAborted();
+    if (toBlock >= remoteFollowBlock) await sleep(params.SECONDS_PER_ETH1_BLOCK * 1000, signal);
```

Comment:
`sleep()` will throw ErrorAbort if the abort signal is already aborted. So the check above it might be unnecessary. Should I delete it or leave as a sanity check?

Reply:
I don't think we should have it. 
IMO we're fine with just the assumption that the signal has not already been thrown if you want to use it to abort the stream.